lib/bap_core_theory/bap_core_theory_program.ml

```diff
@@ -20,6 +20,14 @@ let word = Knowledge.Domain.optional "word"
     ~equal:Bitvec.equal
     ~inspect:Bitvec_sexp.sexp_of_t
 
+let bytes = Knowledge.Domain.optional "bytes"
+    ~equal:String.equal
+    ~inspect:(fun s ->
+        Sexp.Atom (
+          String.to_list s |> List.map ~f:(fun c ->
+              Format.sprintf "%02X" @@ Char.to_int c) |>
+          String.concat ~sep:" "))
+
```

Comment:
Instead of `to_list + map + concat` we can use `String.concat_map`, e.g.,

```suggestion
let hexcode = Knowledge.Domain.optional "hexcode"
    ~equal:String.equal
    ~inspect:(Fn.compose sexp_of_string @@
              String.concat_map ~sep:" " ~f:(fun c ->
                  sprintf "%02x" @@ Char.to_int c))
```

Reply:
👍